lambda_functions/delivery_status_processor_lambda/delivery_status_processor_lambda.py

```diff
@@ -34,12 +39,57 @@
     logger.warning("Invalid log level specified, defaulting to INFO")
 
 
+def get_twilio_token():
+    try:
+        if TWILIO_AUTH_TOKEN_SSM_NAME == 'unit_test':
+            return 'bad_twilio_auth'
+        ssm_client = boto3.client("ssm", "us-gov-west-1")
+
+        response = ssm_client.get_parameter(
+            Name=TWILIO_AUTH_TOKEN_SSM_NAME,
+            WithDecryption=True
+        )
+        return response.get("Parameter").get("Value")
+    except Exception as e:
+        logger.error('Failed to retrieve Twilio Auth with: %s', e)
+        return None
+
+
+auth_token = get_twilio_token()
+
+
+# Duplicated in vetext_incoming_forwarder.
+def validate_twilio_event(event):
+    logger.info("validating twilio delivery event")
+
+    try:
+        signature = event["headers"].get("x-twilio-signature", "")
+
+        validator = RequestValidator(auth_token)
+        uri = f"https://{event['headers']['host']}/vanotify/sms/deliverystatus"
+        decoded = base64.b64decode(event.get("body")).decode()
+        params = parse_qs(decoded)
+        params = {k: v[0] for k, v in sorted(params.items())}
+        return validator.validate(
+            uri=uri,
+            params=params,
+            signature=signature
+        )
+    except Exception as e:
+        logger.error("Error validating request origin: %s", e)
+        return False
+
+
 def delivery_status_processor_lambda_handler(event: any, context: any):
     """this method takes in an event passed in by either an alb.
     @param: event   -  contains data pertaining to an sms delivery status from the external provider
-    @param: context -  contains information regarding information
-        regarding what triggered the lambda (context.invoked_function_arn).
+    @param: context -  AWS context sent by ALB to all events. Over ridden by unit tests as skip trigger.
     """
+    try:
+        if "sec-datadog" in event["headers"]:
+            return {"statusCode": 200}
+    except Exception as e:
+        logger.debug("Passing on issue with synthetic test payload: %s", e)
```

Comment:
```suggestion
        logger.info("Passing on issue with synthetic test payload: %s", e)
```
We should never see this log statement, but if we do enter this area, we should be aware of it. Made it `info` because this is not an issue with production, it is an issue with the configuration of the Synthetic montior and we have alerting in Datadog for that.